### DISCUSSION TITLE
**NEW**  Overseerr LXC

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#1656] [Announcements]
## Overseerr LXC

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/overseerr.sh)"
```

Creates an optimized production build

**Comment by Scales82** [comment]
I asked and received! THANK YOU

**Comment by towerhand** [comment]
Thanks. The script worked like a charm. 

What's the best way to update Overseerr?

**Reply by tteck** [reply]
Execute `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/overseerr.sh)"` in the Overseerr console.

**Reply by towerhand** [reply]
Thank you for adding the update script.

**Comment by nfribeiro** [comment]
Thank You. I installed in a docker lxc, now I will revert to its own lxc container.

**Comment by Drallas** [comment]
Thanks for this LXC script!

I noticed it's on the develop branch.
![Screenshot 2023-08-15 at 14 19 34](https://github.com/tteck/Proxmox/assets/24792888/1338f774-c183-483e-a244-30d35b352d1d) is there a way to change this? I looked in `/opt/overseerr/config/settings.json` but don't see option to change it there.

I don't like the version to be displayed in a 'funky' format.
![Screenshot 2023-08-15 at 14 21 49](https://github.com/tteck/Proxmox/assets/24792888/d0d214b6-9ebf-4268-a581-a9867d83af1e)

**Reply by tteck** [reply]
If you wish to install the "master" branch (currently version 1.33.2), follow the steps outlined below.  It will be a fresh installation, and the "funky" `Keep it up! 👍🏻` will still be there, it's their design choice. 🤷🏻‍♂️ 

```
systemctl stop overseerr.service
rm -rf /opt/overseerr
git clone -q -b master https://github.com/sct/overseerr.git /opt/overseerr
cd /opt/overseerr
yarn install
yarn build
systemctl start overseerr.service
```

**Reply by Drallas** [reply]
Tnx, ran it with: 

```
systemctl stop overseerr.service
cp /opt/overseerr/config/settings.json ~/.
rm -rf /opt/overseerr
git clone -q -b master https://github.com/sct/overseerr.git /opt/overseerr
cd /opt/overseerr
yarn install
yarn build
cp ~/settings.json /opt/overseerr/config/
systemctl start overseerr.service
```

To preserve settings, unfortunately users and their settings are gone, since they (must) be stored in the database (saving the db folder also next time).

Notes:
1. Perhaps it could be an option to add to the installation script if a Develop or Stable(beta) version need to be installed or to switch when updating. 
2. My previous Docker container from linuxserver.io had the version info displayed correctly. 🧐

**Reply by tteck** [reply]
You could backup and restore the entire `config` directory.

```
systemctl stop overseerr.service
cp -R /opt/overseerr/config config-backup
rm -rf /opt/overseerr
git clone -q -b master https://github.com/sct/overseerr.git /opt/overseerr
cd /opt/overseerr
yarn install
yarn build
cd ~
cp -R config-backup/* /opt/overseerr/config
rm -rf config-backup
systemctl start overseerr.service
```
That gets everything (`db`  `logs`  `settings.json`)

**Reply by tonygauderman** [reply]
I'm assuming re-running your update script after switching to master would pull the development branch again?

**Comment by bobloadmire** [comment]
on installation i'm getting this:

```
> yarn install v1.22.22
> [1/4] Resolving packages...
> [2/4] Fetching packages...
> info There appears to be trouble with your network connection. Retrying...
> info There appears to be trouble with your network connection. Retrying...
> info There appears to be trouble with your network connection. Retrying...
>  \[############################################################################################################################################################----------------------------------------------------------------------------------------------------------------------------------------------------------------------] 878/181
> info There appears to be trouble with your network connection. Retrying...
> info There appears to be trouble with your network connection. Retrying...
>  \[############################################################################################################################################################################################################################################################---------------------------------------------------------------------] 1423/181
> error Error: https://registry.yarnpkg.com/date-fns/-/date-fns-2.29.3.tgz: ESOCKETTIMEDOUT
>     at ClientRequest.<anonymous> (/usr/lib/node_modules/yarn/lib/cli.js:142037:19)
>     at Object.onceWrapper (node:events:632:28)
>     at ClientRequest.emit (node:events:518:28)
>     at TLSSocket.emitRequestTimeout (node:_http_client:845:9)
>     at Object.onceWrapper (node:events:632:28)
>     at TLSSocket.emit (node:events:530:35)
>     at Socket._onTimeout (node:net:589:8)
>     at listOnTimeout (node:internal/timers:573:17)
>     at process.processTimers (node:internal/timers:514:7)
> info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
>  -
> [ERROR] in line 43: exit code 0: while executing command $STD yarn install
```

**Comment by Lockie85** [comment]
Hey @tteck 

If I run the command:
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/overseerr.sh)"

Will it update the LXC? As when I do it, it says:
This will create a New Overseerr LXC

In addition, do I need to shutdown the current LXC to update?

Lastly, will running the command keep all settings/data?

**Reply by Lockie85** [reply]
Ignore me, I've just seen this on the site:
To Update Overseerr, run the command below (or type update) in the LXC Console.

**Comment by kelmer44** [comment]
My pve is getting stuck in `Installing Overseerr (Patience)  `. The LXC seems to be running but the server isnt available and when I open the console Im asked for overseerr login :/

**Reply by tteck** [reply]
> Patience

**Reply by towerhand** [reply]
You can try increasing the ram and cpu.
